yt_dlp/extractor/nobelprize.py

```diff
@@ -1,59 +1,62 @@
+            'description': 'md5:1a2d8a6ca80c88fb3b9a326e0b0e8e43',
+            'duration': 1560.0,
+            'thumbnail': r're:https?://www\.nobelprize\.org/images/.+\.jpg',
+            'timestamp': 1504883793,
+            'upload_date': '20170908',
         },
-    }
+    }, {
+        'url': 'https://mediaplayer.nobelprize.org/mediaplayer/?qid=12693',
+        'info_dict': {
+            'id': '12693',
+            'ext': 'mp4',
+            'title': 'Nobel Lecture by Peter Higgs',
+            'description': 'md5:9b12e275dbe3a8138484e70e00673a05',
+            'duration': 1800.0,
+            'thumbnail': r're:https?://www\.nobelprize\.org/images/.+\.jpg',
+            'timestamp': 1504883793,
+            'upload_date': '20170908',
+        },
+    }]
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
+        url = update_url(url, netloc='mediaplayer.nobelprize.org')
+        video_id = traverse_obj(parse_qs(url), (
+            ('id', 'qid'), 0, any, {require('video ID')}))
         webpage = self._download_webpage(url, video_id)
-        media = self._parse_json(self._search_regex(
-            r'(?s)var\s*config\s*=\s*({.+?});', webpage,
-            'config'), video_id, js_to_json)['media']
-        title = media['title']
-
-        formats = []
-        for source in media.get('source', []):
-            source_src = source.get('src')
-            if not source_src:
-                continue
-            ext = mimetype2ext(source.get('type')) or determine_ext(source_src)
-            if ext == 'm3u8':
-                formats.extend(self._extract_m3u8_formats(
-                    source_src, video_id, 'mp4', 'm3u8_native',
-                    m3u8_id='hls', fatal=False))
-            elif ext == 'f4m':
-                formats.extend(self._extract_f4m_formats(
-                    update_url_query(source_src, {'hdcore': '3.7.0'}),
-                    video_id, f4m_id='hds', fatal=False))
-            else:
-                formats.append({
-                    'url': source_src,
-                })
 
         return {
             'id': video_id,
-            'title': title,
-            'description': get_element_by_attribute('itemprop', 'description', webpage),
-            'duration': int_or_none(media.get('duration')),
-            'formats': formats,
+            'title': self._html_search_meta('caption', webpage),
+            'description': traverse_obj(webpage, (
+                {find_element(tag='span', attr='itemprop', value='description')}, {clean_html})),
+            'duration': parse_duration(self._html_search_meta('duration', webpage)),
+            **traverse_obj(next(self._yield_json_ld(webpage, video_id)), {
+                'thumbnail': ('thumbnail_url', {lambda x: self._proto_relative_url(x)}, {url_or_none}),
+                'timestamp': ('uploadDate', {unified_timestamp}),
+                'url': ('contentUrl', {url_or_none}),
+            }),
```

Comment:
is there a reason why we can't do this?
```suggestion
            **self._search_json_ld(webpage, video_id),
```
(though you'd probably want to unpack this above the other fields so as not to overwrite them?)

Reply:
> is there a reason why we can't do this?

In this case, `_search_json_ld` doesn’t return thumbnails

Reply:
Actually, why don't we just fix this in `InfoExtractor`?

Reply:
The common JSON LD code was written before `traverse_obj` supported dict key traversal or inline transformations. Let's take advantage of those:
```diff
diff --git a/yt_dlp/extractor/common.py b/yt_dlp/extractor/common.py
index d5607296d..eb9abcdeb 100644
--- a/yt_dlp/extractor/common.py
+++ b/yt_dlp/extractor/common.py
@@ -1675,9 +1675,9 @@ def extract_video_object(e):
                 'ext': mimetype2ext(e.get('encodingFormat')),
                 'title': unescapeHTML(e.get('name')),
                 'description': unescapeHTML(e.get('description')),
-                'thumbnails': [{'url': unescapeHTML(url)}
-                               for url in variadic(traverse_obj(e, 'thumbnailUrl', 'thumbnailURL'))
-                               if url_or_none(url)],
+                'thumbnails': traverse_obj(e, (('thumbnailUrl', 'thumbnailURL', 'thumbnail_url'), {
+                    'url': ({unescapeHTML}, {self._proto_relative_url}, {url_or_none}),
+                })),
                 'duration': parse_duration(e.get('duration')),
                 'timestamp': unified_timestamp(e.get('uploadDate')),
                 # author can be an instance of 'Organization' or 'Person' types.
```